File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        //1.use string equals method,not ==
        //2.prefix notation is polish notation
        //3.postfix notation is reverse polish notation
        Stack<Integer> stack=new Stack<Integer>();
        for(String token : tokens){
            if(token.equals("+")){
                stack.push(stack.pop()+stack.pop());
            }
            else if(token.equals("*")){
                stack.push(stack.pop()*stack.pop());
            }
            else if(token.equals("-")){
                int a=stack.pop();
                int b=stack.pop();
                stack.push(b-a);
            }
            else if(token.equals("/")){
                int a=stack.pop();
                int b=stack.pop();
                stack.push(b/a);
            }
            else{
                stack.push(Integer.valueOf(token));
                //OR
                //stack.push(Integer.parseInt(token));
            }
        }
        return stack.pop();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack=[]
        for i in range(len(tokens)):
            if tokens[i]=="+":
                first=stack.pop()
                second=stack.pop()
                stack.append(first+second)
            elif tokens[i]=="*":
                first=stack.pop()
                second=stack.pop()
                stack.append(first*second)
            elif tokens[i]=="-":
                first=stack.pop()
                second=stack.pop()
                stack.append(second-first)
            elif tokens[i]=="/":
                first=stack.pop()
                second=stack.pop()
                stack.append(int(second/first))
            else:
                stack.append(int(tokens[i]))
        return stack.pop()
        